apps/shipments/views/shipment.py

```diff
@@ -178,6 +179,38 @@ def tracking(self, request, version, pk):
 
         raise NotFound("No tracking data found for Shipment.")
 
+    @method_decorator(cache_page(60 * 60, remember_all_urls=True))  # Cache responses for 1 hour
+    @action(detail=True, methods=['get'], permission_classes=(IsOwnerOrShared,))
+    def telemetry(self, request, version, pk):
```

Comment:
Most of the logic in the original tracking data view that you copied here was GeoJSON-specific optimizations. I think you can simplify this view a lot, and possibly use filtersets to handle all of your conditional filtering/query_params parsing.

Reply:
Please revisit this, I'm not sure you've taken a close enough look. I am pretty sure I still see room for improvement here